plan/validator.go

```diff
@@ -444,12 +444,27 @@ func checkColumn(colDef *ast.ColumnDef) error {
 				return types.ErrIllegalValueForType.GenByArgs(types.TypeStr(tp.Tp), str)
 			}
 		}
+	case mysql.TypeDate:
```

Comment:
`NOW` function is invalid for other types too, it's valid for only `DATETIME` and `TIMESTAMP`.